parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -384,6 +388,22 @@ configureContainerdRegistryHost() {
 EOF
 }
 
+configureContainerdLegacyMooncakeMcrHost() {
+    LEGACY_MCR_REPOSITORY_BASE="mcr.azk8s.cn"
+    CONTAINERD_CONFIG_REGISTRY_HOST_MCR="/etc/containerd/certs.d/${LEGACY_MCR_REPOSITORY_BASE}/hosts.toml"
+    mkdir -p "$(dirname "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}")"
+    touch "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}"
+    chmod 0644 "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}"
+
+    TARGET_MCR_REPOSITORY_BASE="mcr.azure.cn"
+    CONTAINER_REGISTRY_URL=$(sed 's@/@/v2/@1' <<< "${TARGET_MCR_REPOSITORY_BASE}/")
```

Comment:
Should we remove the /v2/ since `mcr.azure.cn` is OCI compliant?

Some context here: `/v2/` is used for NI since ACR artifact cache doesn't support [prefixed repository mapping](https://dev.azure.com/msazure/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/678808/image-pull-through-mirror-config?anchor=container-config-doesn%27t-support-prefix), `/some-prefix/...` -> `mcr.microsoft.com/some-prefix/...`.

Since such feature doesn't need to supported for MAR, I think we don't need the /v2/ here